internal/resolution/variable_sources/required_package/required_package.go

```diff
@@ -0,0 +1,66 @@
+package required_package
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/operator-framework/deppy/pkg/deppy"
+	"github.com/operator-framework/deppy/pkg/deppy/constraint"
+	"github.com/operator-framework/deppy/pkg/deppy/input"
+
+	olmentity "github.com/operator-framework/operator-controller/internal/resolution/variable_sources/entity"
+	"github.com/operator-framework/operator-controller/internal/resolution/variable_sources/utils/predicates"
+	"github.com/operator-framework/operator-controller/internal/resolution/variable_sources/utils/sort"
+)
+
+type RequiredPackageVariable struct {
+	*input.SimpleVariable
+	bundleEntities []*olmentity.BundleEntity
+}
+
+func (r *RequiredPackageVariable) BundleEntities() []*olmentity.BundleEntity {
+	return r.bundleEntities
+}
+
+func NewRequiredPackageVariable(packageName string, bundleEntities []*olmentity.BundleEntity) *RequiredPackageVariable {
+	id := deppy.IdentifierFromString(fmt.Sprintf("required package %s", packageName))
+	var entityIDs []deppy.Identifier
+	for _, bundle := range bundleEntities {
+		entityIDs = append(entityIDs, bundle.ID)
+	}
+	return &RequiredPackageVariable{
+		SimpleVariable: input.NewSimpleVariable(id, constraint.Mandatory(), constraint.Dependency(entityIDs...)),
+		bundleEntities: bundleEntities,
+	}
+}
+
+var _ input.VariableSource = &RequiredPackageVariableSource{}
```

Comment:
Not sure I get this construct?

Reply:
It's ensuring that `RequiredPackageVariableSource` correctly implements `input.VariableSource`: https://go.dev/doc/effective_go#blank_implements